beacon_node/beacon_chain/src/beacon_chain_builder.rs

```diff
@@ -210,22 +210,19 @@ fn interop_genesis_state<T: EthSpec>(
         .collect::<Vec<_>>();
 
     let mut proofs = vec![];
-    for i in 1..=deposit_root_leaves.len() {
-        // Note: this implementation is not so efficient.
-        //
-        // If `MerkleTree` had a push method, we could just build one tree and sample it instead of
-        // rebuilding the tree for each deposit.
-        let tree = MerkleTree::create(
-            &deposit_root_leaves[0..i],
-            spec.deposit_contract_tree_depth as usize,
-        );
+    let depth = spec.deposit_contract_tree_depth as usize;
+    let mut tree = MerkleTree::create(&[], depth);
+    for (i, deposit_leave) in deposit_root_leaves.iter().enumerate() {
```

Comment:
Super minor nit: `deposit_leave` -> `deposit_leaf`.

I don't mind if this isn't fixed though, the meaning is clear.

Reply:
** coughs ** yes that was intentional ofc... fixed it ;)